Updates to hardware requirements

jasonblais opened the pull request.

jasonblais commented on source/deployment/scaling.rst:
```diff
@@ -4,32 +4,34 @@ Scaling for Enterprise
 ^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Organizations less than 500 users can typically run on a single server. 
+Organizations less than 2,000 users can typically run on a single server. 
```
Is the number "2000" correct?

jwilander replied:
It really depends on usage. Pre-release is a 4,500 user server and I would be very comfortable running it on a single 2 vCPU 4GB RAM machine.

jasonblais commented on source/install/requirements.rst:
```diff
@@ -101,60 +101,31 @@ Hardware Sizing for Team Deployments
--  250-500 users - 2 vCPUs/cores, 4 GB RAM, and 45-90 GB storage
--  500-1,000 users - 4 vCPUs/cores, 8 GB RAM, and 90-180 GB storage
--  1,000-2,000 users - 4-8 vCPUs/cores, 16-32 GB RAM, and 180-360 GB storage
+-  1 - 1,000 users - 1 vCPU/cores, 2 GB RAM, and 45-90 GB storage
```
I'm not sure if the storage recommendations are relevant, but we have a note `Storage recommendation is based on storing 3 years of archives with moderate file sharing.`.

Wondering if we should instead ask the admin to estimate file storage requirements based on file sharing within their team. I feel it's highly dependent on the use cases (e.g. a design firm probably requires a lot more file storage capacity compared to a dev ops team)

jwilander replied:
0/5 the storage will also only matter if they're not using minio or S3

jasonblais commented on source/install/requirements.rst:
```diff
@@ -101,60 +101,31 @@ Hardware Sizing for Team Deployments
-For Mattermost Enterprise Edition deployments, an open source load testing framework is available to simulate usage: https://github.com/mattermost/mattermost-load-test
+  1. Memory requirements are largely driven by peak file sharing activity. Recommendation is based on default 50 MB max file size, which can be adjusted from the System Console. Changing this number may change memory requirements. 
+  2. Storage recommendation is based on storing 3 years of archives with moderate file sharing.
+  3. Solid state drives (SSD) can be used in place of disk storage for higher concurrency.
```
Is point 3 relevant?

jwilander replied:
I think we can omit it

jasonblais commented on source/install/requirements.rst:
```diff
@@ -101,60 +101,31 @@ Hardware Sizing for Team Deployments
+.. note::
 
-- Regular hard drives can be used in place of solid-state hard drives if having top performance is not a priority. If using a mix of HDD and SSD drives, the greatest performance gain would come from using SSD in the database server.
+  Regular hard drives (HDD) can be used in place of solid-state hard drives (SSD) if having top performance is not a priority. If using a mix of HDD and SSD drives, the greatest performance gain comes from using SSD in the database server.
```
Is this note a duplication of https://github.com/mattermost/docs/pull/2302/files#diff-12b8342499d6b7515730e432a66bcda9R113

jwilander replied:
Essentially

jwilander commented:
I'd suggest we also add a note saying that these recommendations are very much ballpark figures and will vary based on usage. Something like: "These recommendations are based off traditional Team Edition deployments and may grow or shrink depending on how active your users are." I'm essentially basing my recommendations off pre-release which is probably typical of a community TE deployment. Someone deploying TE to a company may see higher usage and need a higher specced machine as a result

jasonblais commented:
Updated based on feedback:

1. Reworded the note about a single machine for less than 2,000 users to convey you can run higher numbers based on usage and server configuration https://github.com/mattermost/docs/pull/2302/files#diff-c075e9d1123114d517e20a9fc92d1b62R12
2. Updated intro of hardware requirements section, and moved notes relevant to both TE and EE to this section
3. Removed storage recommendations, given we have a "Alternate Storage Calculator" at the bottom of this page
4. Removed note about SSDs
5. Made it more clear these recommendations highly depend on usage

jwilander commented on source/deployment/scaling.rst:
```diff
@@ -4,32 +4,34 @@ Scaling for Enterprise
 ^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Organizations less than 500 users can typically run on a single server. 
+Organizations can typically run Mattermost on a single server with up to 2,000 users, though higher numbers have been observed based on different usage and server configurations.
```
0/5 but maybe `higher numbers` to `more users` just to be a bit more clear
